examples/JordanGradient.ipynb

```diff
@@ -0,0 +1,354 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# reference for computing error\n",
+    "def analytic_gradient(x):\n",
+    "    if x < 0:\n",
+    "        g = -1 * POTENTIAL_BARRIER_VALUE\n",
+    "    elif x >= 1:\n",
+    "        g = POTENTIAL_BARRIER_VALUE\n",
+    "    else:\n",
+    "        g = x - DOMAIN_SHIFT  \n",
+    "    return g"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we will estimate the gradient at 4 random domain values.  So that this example is deterministic, we specify the random seed as 99.\n",
+    "\n",
+    "We also need to set region over which our sampling region is approximately linear (PERTURBATION_MAGNITUDE), and the number of times to measure the system (N_MEASUREMENTS)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# compute gradient at 4 random points [0, 1). \n",
+    "np.random.seed(99)\n",
+    "x_scan = np.random.rand(4)\n",
+    "\n",
+    "# set constants\n",
+    "PERTURBATION_MAGNITUDE = .001\n",
+    "N_MEASUREMENTS = 50"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Because we are simulating connecting to the quantum computer, here we have mock the response.  \n",
+    "\n",
+    "These results were generated from running the notebook live.\n",
+    "\n",
+    "If you are running live, ignore the below cell, and simply run this code:\n",
+    "\n",
+    "```python\n",
+    "from pyquil.api import QVMConnection\n",
+    "qvm = QVMConnection()\n",
+    "```\n",
+    "\n",
+    "Then, you would add ```cxn=qvm``` to the argument list when you call ```potential_gradient```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "binary_p4 = {\n",
+    "    x_scan[0]: [1, 0, 0, 0],\n",
+    "    x_scan[1]: [0, 0, 0, 0],\n",
+    "    x_scan[2]: [1, 1, 0, 0],\n",
+    "    x_scan[3]: [0, 0, 1, 0]\n",
+    "}\n",
+    "\n",
+    "binary_p10 = {\n",
+    "    x_scan[0]: [0, 0, 0, 1, 1, 0, 1, 0, 0, 0],\n",
+    "    x_scan[1]: [0, 1, 1, 0, 0, 0, 0, 0, 0, 0],\n",
+    "    x_scan[2]: [1, 1, 1, 0, 0, 1, 0, 1, 0, 0],\n",
+    "    x_scan[3]: [0, 0, 0, 0, 1, 1, 1, 1, 0, 0]\n",
+    "}\n",
+    "\n",
+    "binary_maps = {\n",
+    "    4: binary_p4,\n",
+    "    10: binary_p10\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def potential_gradient(x, h, n_measurements, precision, precision_map, cxn=None):\n",
+    "    \n",
+    "    # if you are running this live, input your qvm or qpu object to cxn\n",
+    "    if not cxn:\n",
```

Comment:
This should be `if cxn is None`, this is from PEP8: https://www.python.org/dev/peps/pep-0008/#programming-recommendations